hail/src/main/scala/is/hail/expr/ir/InferPType.scala

```diff
@@ -5,6 +5,66 @@ import is.hail.expr.types.virtual.{TNDArray, TVoid}
 import is.hail.utils._
 
 object InferPType {
+
+  def clearPTypes(x: IR): Unit = {
+    x._pType2 = null
+    x.children.foreach { c => clearPTypes(c.asInstanceOf[IR]) }
+  }
+
+  // does not unify physical arg types if multiple nested seq/init ops appear; instead takes the first. The emitter checks equality.
+  def computePhysicalAgg(virt: AggStateSignature, initsAB: ArrayBuilder[RecursiveArrayBuilderElement[InitOp]],
```

Comment:
Sorry, not clear headed enough to evaluate these right now, largely unfamiliar with these nodes. If you're confident that they work, I'll accept and make a note to myself to go over later. No need to block you I think.